src/test/java/org/folio/marc4ld/mapper/field502/Bibframe2Marc502IT.java

```diff
@@ -0,0 +1,85 @@
+package org.folio.marc4ld.mapper.field502;
+
+import static org.folio.ld.dictionary.PropertyDictionary.DEGREE;
+import static org.folio.ld.dictionary.PropertyDictionary.DISSERTATION_ID;
+import static org.folio.ld.dictionary.PropertyDictionary.DISSERTATION_NOTE;
+import static org.folio.ld.dictionary.PropertyDictionary.DISSERTATION_YEAR;
+import static org.folio.ld.dictionary.PropertyDictionary.LABEL;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.INSTANCE;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.WORK;
+import static org.folio.marc4ld.mapper.test.TestUtil.loadResourceAsString;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.assertj.core.api.AssertionsForClassTypes;
+import org.folio.ld.dictionary.PredicateDictionary;
+import org.folio.ld.dictionary.PropertyDictionary;
+import org.folio.ld.dictionary.ResourceTypeDictionary;
+import org.folio.ld.dictionary.model.Resource;
+import org.folio.marc4ld.mapper.test.MonographTestUtil;
+import org.folio.marc4ld.mapper.test.SpringTestConfig;
+import org.folio.marc4ld.service.ld2marc.Bibframe2MarcMapperImpl;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@EnableConfigurationProperties
+@SpringBootTest(classes = SpringTestConfig.class)
+class Bibframe2Marc502IT {
+
+  @Autowired
+  private Bibframe2MarcMapperImpl bibframe2MarcMapper;
+
+  @Test
+  @Disabled // TODO - Discuss with team tomorrow
+  void shouldMapField502() {
```

Comment:
Hi @PBobylev @aleksei-pronichev  @askhat-abishev,

I noticed one problem with graph -> marc transformation. 

Let us consider this graph

<img width="886" alt="image" src="https://github.com/FOLIO-EIS/lib-linked-data-marc4ld/assets/121239864/3aaccf34-a9b3-43ba-bb3b-a819a5347a94">

When MARC is derived from this graph, the expected MARC is as below

> 502 $b PHD $c XYZ University
> 502 $b Doctor $c ABC University

However, the actual MARC is as below
> 502 $b PHD
> 502 $c XYZ University
> 502 $b Doctor
> 502 $c ABC University

From this MARC, we can't correlate between a degree and the corresponding granting institution.

Reply:
@pkjacob DataFieldCombiner is ok for me. It was created exactly for this cases;
pluses
1. just need to add 1 class, implemented the interface
2. no changes in global logic